free-2.md

```diff
@@ -8,44 +8,103 @@ good way to meet their needs.
+being too redundent and wanting a way to specify weights for artists and bands
+that I wanted to listen to.
+
+I ran it by a few friends to get their feedback and they thought it would be a
+neat idea to pursue and implement.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Intelligent and controllable playlist creation.
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
-
+_How might the user interact with the language? What does programming look
+like? Why is this the right way to interact with the problem domain?_
+
+Since this langugae would target people with a variety of backgrounds in programming&mdash;
+mostly people without experience&mdash;I imagine a Scratch-like visual language.
+
+The blocks however, would be more tailored to music playlists. Some examples include:
++ artist: specifies an artist of a song
++ band: specifies a band
++ song: specifies a song name directly
++ not: negates an expression so you could express things along the lines of
+       "I don't want to listen to ABBA".
++ redundency: number of songs belonging to an entity and a time in which you
+  would like to hear them
++ divergence: specify the divergence factor from the current song
++ bpm: specifies beats per minute of the song
+
+While I think it may be easier to implement a DSL that looks like a programming
+langugae, a visual language would be easier to learn and an artistic
+component to the playlist. Each playlist would, through its visual prgamming,
+have its own artwork/fingerprint.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+A program will begin at the start node and work its way down through the flow
+of the playlist making choices for next songs based on the current node it is
+in. A song will play (or be queued up) and retrieved from a library of music.
+With each step, the computer will indicate which block it is on making it easier
+for users to adjust it. Ideally this DSL would have access to Spotify or
+another music service where we have a large volume of choices and freedom to
+play them.
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to make an ordered playlist like the traditional method and
+it should be easy to pick random songs or albums. New discovery should be easy.
+It should not be able to cut tracks; it should be able to play them back at differnent
+volumes and maybe speed up or slow them down. It should not be a DJ program.
+It should be possible to reproduce a playlist through seeding random generators
+so I could share a playlist with you that you could then edit to your own liking.
 
```

Comment:
If you implement sharing, please do something like [this](http://what3words.com). Users hate typing out hex (see Mario Maker). Typing out words is easy.